docs/writing_a_builder.md

```diff
@@ -31,6 +36,43 @@ will be passed as a primary input, and each build step may produce two assets.
 For the primary input `some_file.txt` the allowed outputs are
 `some_file.txt.foo` and `some_file.txt.bar`.
 
+### Capture groups
+
+Builders can declare more complex inputs and outputs by using a capture group
+in their build input. Capture groups allow builders to move files across
```

Comment:
[nit] I worry that "files" could be misconstrued to include the input.

```suggestion
in their build input. Capture groups allow builders to move outputs across
```

Reply:
Or maybe
```suggestion
in their build input. Capture groups allow builders to copy files across
```

Reply:
I went with "Capture groups can be used to write outputs in a different directory than the primary input." which stays closer to build's common terminology.